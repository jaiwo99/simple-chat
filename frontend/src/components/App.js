require('normalize.css/normalize.css');
require('../styles/App.css');

import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
import React from 'react';
import Store from '../services/LocalStorage';

import Login from './Login';
import Chat from './Chat';
import Event from './Event';
import ChatService from './Chat.serivce';

export default class App extends React.Component {

  constructor(props) {
    super(props);

    this.store = new Store();
    this.chatService = new ChatService();
    this.state = {};
  }

  componentDidMount() {
    this.reload();
  }

  reload() {
    let name = this.store.read();
    if (name) {
      this.author = name;
      this.setState({
        showChat: true
      });
    } else {
      this.setState({
        showChat: false
      });
    }
  }

  render() {
    if (this.state.showChat) {
      return (
        <div className="wrapper">
          <Chat author={this.author} chatService={this.chatService}/>
          <Event author={this.author} chatService={this.chatService}/>
        </div>
      )
    } else {
      return (<Login store={this.store} reloadFn={this.reload.bind(this)}/>)
    }
  }
}
