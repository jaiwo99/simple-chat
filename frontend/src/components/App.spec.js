import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

describe('App Component', () => {

  test('should render login component', () => {
    localStorage.clear();

    const component = renderer.create(
      <App />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(tree.props.className).toBe('login');

  });

  test('should ask author name from local storage', () => {
    localStorage.clear();

    renderer.create(
      <App />
    );

    expect(localStorage.getItem).toHaveBeenLastCalledWith('CHAT_AUTHOR');
  });
});

