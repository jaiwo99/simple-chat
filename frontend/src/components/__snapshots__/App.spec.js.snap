// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render login component 1`] = `
<div
  className="container"
>
  <h1
    className="jumbotron"
  >
    Simple Chat
  </h1>
  <div
    className="login"
  >
    <h3>
      Please enter your name here:
    </h3>
    <form
      className="form-inline"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="sr-only"
          htmlFor="author"
        >
          Enter your name here: 
        </label>
        <input
          className="form-control"
          id="author"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
        
      <div
        className="form-group"
      >
        <button
          className="btn btn-primary"
          type="submit"
        >
          Enter
        </button>
      </div>
    </form>
  </div>
</div>
`;
